### PR TITLE
Add support for isolation to the nftables driver

### PR DESCRIPTION
**Pull request by mheon**
Replicates the current iptables approach (3 isolation chains, with support for strict and normal isolation, or no isolation at all). No significant changes from iptables implementation.

### CODE REVIEW

**Comment by mheon** [comment]
Still needs tests, so WIP

**Comment by mheon** [comment]
Enabled the test. Seems to work in local testing.

**Comment by mheon** [comment]
@Luap99 PTAL when you get back

File: src/firewall/nft.rs [hunk 1]
```diff
@@ -1058,6 +1206,26 @@ fn get_rule_matcher_jump_to(jump_target: String) -> Box<dyn Fn(&schema::Rule) ->
     })
 }
 
+/// Make a closure that matches any rule that tests for a match to a given bridge interface.
+fn get_rule_matcher_bridge(bridge: String) -> Box<dyn Fn(&schema::Rule) -> bool> {
+    Box::new(move |r: &schema::Rule| -> bool {
```
Comment (Luap99) [inline src/firewall/nft.rs:1211]: we do not need the box here and we can also get rid of the bridge String with the use of a lifetime.
```diff
diff --git a/src/firewall/nft.rs b/src/firewall/nft.rs
index b123a2a..9c75380 100644
--- a/src/firewall/nft.rs
+++ b/src/firewall/nft.rs
@@ -246,7 +246,7 @@ impl firewall::FirewallDriver for Nftables {
             ));
         }
 
-        let match_our_bridge = get_rule_matcher_bridge(network_setup.bridge_name.clone());
+        let match_our_bridge = get_rule_matcher_bridge(&network_setup.bridge_name);
 
         // If and only if isolation is enabled: add isolation chains.
         // Some isolation rules are shared. Other rules are specific to one type
@@ -506,7 +506,7 @@ impl firewall::FirewallDriver for Nftables {
             }
         }
 
-        let match_our_bridge = get_rule_matcher_bridge(tear.config.bridge_name.clone());
+        let match_our_bridge = get_rule_matcher_bridge(&tear.config.bridge_name);
 
         let mut isolation_rules: Vec<schema::Rule> = Vec::new();
         isolation_rules.append(&mut get_matching_rules_in_chain(
@@ -1207,13 +1207,13 @@ fn get_rule_matcher_jump_to(jump_target: String) -> Box<dyn Fn(&schema::Rule) ->
 }
 
 /// Make a closure that matches any rule that tests for a match to a given bridge interface.
-fn get_rule_matcher_bridge(bridge: String) -> Box<dyn Fn(&schema::Rule) -> bool> {
-    Box::new(move |r: &schema::Rule| -> bool {
+fn get_rule_matcher_bridge(bridge: &str) -> impl '_ + Fn(&schema::Rule) -> bool {
+    move |r: &schema::Rule| -> bool {
         for statement in &r.expr {
             match statement {
                 stmt::Statement::Match(m) => match &m.right {
                     expr::Expression::String(s) => {
-                        if *s == bridge {
+                        if s == bridge {
                             return true;
                         }
                     }
@@ -1223,7 +1223,7 @@ fn get_rule_matcher_bridge(bridge: String) -> Box<dyn Fn(&schema::Rule) -> bool>
             }
         }
         false
-    })
+    }
 }
 
 /// Find all rules in the given chain which match the given closure (true == include).
```

**Review by Luap99 — COMMENTED** [review]
mostly LGTM, one comment which should make the code a bit simpler IMO

I also assume this has the same problem as described in https://github.com/containers/netavark/issues/709 but that is not a blocker given it would be consistent with iptables and we can fix it at a later point

File: src/firewall/nft.rs (hunk 1 above)
Comment (mheon) [reply src/firewall/nft.rs]: Changes pushed, TY

**Review by Luap99 — APPROVED** [review]
/lgtm
/hold

**Pull request merged**